spacy/ml/tb_framework.pyx

```diff
@@ -249,9 +249,11 @@ cdef list _parse_batch(CBlas cblas, TransitionSystem moves, StateC** states,
     cdef np.ndarray step_actions
 
     scores = []
-    while sizes.states >= 1:
+    while sizes.states >= 1 and (actions is None or actions):
         step_scores = numpy.empty((sizes.states, sizes.classes), dtype="f")
         step_actions = actions[0] if actions is not None else None
+        assert step_actions is None or step_actions.size == sizes.states, \
```

Comment:
Would it be easy to come up with a straight-forward (failing) test to catch this assertion? Given how the violation of this invariant can lead to segfaults, it might not be a bad idea to have a test or two for it.

Reply:
Added. I would also like to fix the segfault that occurs when feeding the wrong actions. But that requires changes to `TransitionSystem`, for which I'll make a PR after this one (to avoid that this PR becomes too unfocused).